firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java

```diff
@@ -133,35 +153,51 @@ public void addIndexEntries(Document document) {
                         row.getInt(2),
                         row.getInt(3));
 
-                List<Value> values = extractFieldValue(document, fieldIndex);
-                if (values == null) return;
-
-                if (Logger.isDebugEnabled()) {
-                  Logger.debug(
-                      TAG,
-                      "Adding index values for document '%s' to index '%s'",
-                      documentKey,
-                      fieldIndex);
-                }
+                addIndexEntry(document, Collections.singletonList(fieldIndex), null);
 
-                Object[] encodeValues = encodeDocumentValues(fieldIndex, values);
-                for (Object encoded : encodeValues) {
-                  // TODO(indexing): Handle different values for different users
-                  db.execute(
-                      "INSERT OR IGNORE INTO index_entries ("
-                          + "index_id, "
-                          + "index_value, "
-                          + "document_name) VALUES(?, ?, ?)",
-                      indexId,
-                      encoded,
-                      documentKey.toString());
-                }
               } catch (InvalidProtocolBufferException e) {
                 throw fail("Invalid index: " + e);
               }
             });
   }
 
+  public int addIndexEntry(
+      Document document,
+      Collection<FieldIndex> fieldIndexes,
+      @Nullable Map<FieldIndex, SnapshotVersion> fieldIndexToSnapshotVersion) {
```

Comment:
You can likely remove this map too if you update the version in the FieldIndex direclty.

Reply:
Removed.